### PR TITLE
[QC-783] Fix service discovery disconnection

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Comment by Barthelemy** [comment]
@knopers8 could you have a look at this PR please ?

**Review by knopers8 — APPROVED** [review]

File: Framework/src/ServiceDiscovery.cxx [hunk 1]
```diff
@@ -120,14 +120,19 @@ void ServiceDiscovery::runHealthServer(unsigned int port)
...
       });
       tcp::socket socket(io_service);
       acceptor.async_accept(socket, [&](boost::system::error_code ec) {
-        if (!ec)
+        if (ec.failed()) {
+          threadInfoLogger << "async_accept error: " << ec.message() << ENDM;
           timer.cancel();
+        }
       });
+      io_service.run();
```
Comment (awegrzyn) [inline Framework/src/ServiceDiscovery.cxx:135]: This need to be run once per process. Maybe it was removed from QC/DPL code at some point? Could you verify?

Comment (awegrzyn) [reply Framework/src/ServiceDiscovery.cxx:135]: It's seems reasonable to have it, do you have any stack traces? Or the only behaviour was that process stopped responding on Consul health checks?

Comment (Barthelemy) [reply Framework/src/ServiceDiscovery.cxx:135]: The only behaviour that I can see is disappearance of the service in Consul.

**Pull request merged**